presto-docs/src/main/sphinx/admin/verifier.rst

```diff
@@ -0,0 +1,257 @@
+        control_query text NOT NULL,
+        control_username varchar(256) DEFAULT NULL,
+        control_password varchar(256) DEFAULT NULL,
+        control_session_properties text DEFAULT NULL,
+        test_catalog varchar(256) NOT NULL,
+        test_schema varchar(256) NOT NULL,
+        test_query text NOT NULL,
+        test_username varchar(256) DEFAULT NULL,
+        test_password varchar(256) DEFAULT NULL,
+        test_session_properties text DEFAULT NULL)
+
+Next, create a ``config.properties`` file:
+
+.. code-block:: none
+
+    source-query.suite=my_suite
+    source-query.database=jdbc:mysql://localhost:3306/my_database?user=my_username&password=my_password
+    control.gateway=jdbc:presto://localhost:8080
+    test.gateway=jdbc:presto://localhost:8081
+    test-id=1
+
+Download :maven_download:`verifier` and rename it to ``verifier``. To run the Verifier:
+
+.. code-block:: none
+
+    chmod +x verifier
+    ./verifier verify config.properties
+
+
+Verifier Procedures
+-------------------
+
+The following steps summarize the workflow of Verifier.
+
+* **Importing Source Queries**
+   * Read the list of source queries (query pairs with configuration) from the MySQL table.
+
+* **Query Pre-processing and Filtering**
+   * Apply overrides to the catalog, schema, username, and password of each query.
+   * Filter queries according to whitelist and blacklist. Whitelist is applied before blacklist.
+   * Filter out queries with invalid syntax.
+   * Filter out queries not supported for validation. ``Select``, ``Insert``, and ``CreateTableAsSelect`` are supported.
+
+* **Query rewriting**
+    * Rewrite queries before execution to ensure that production data is not modified.
+    * Rewrite ``Select`` queries to ``CreateTableAsSelect``
+       * Column names are determined by running the ``Select`` query with ``LIMIT 0``.
+       * Artificial name are used for unnamed columns.
+    * Rewrite ``Insert`` and ``CreateTableAsSelect`` queries to have their table names replaced.
+       * Construct a setup query to create the table necessary for a ``Insert`` query.
+
+* **Query Execution**
+    * Conceptually, Verifier is configured with a control cluster and a test cluster. However, they
+      may be pointed to the same Presto cluster for certain tests.
+    * For each source query, execute the following queries in order.
+        * Control setup queries
+        * Control query
+        * Test setup queries
+        * Test query
+        * Control and Test teardown
+    * Queries are subject to timeouts and retries.
+        * Cluster connection failures and transient Presto failures are retried.
+        * Query retries may conceal reliability issues. All occurred Presto query failures, including the retries ones, are recorded.
+    * Certain query failures are automatically submitted for re-validation, such as partition dropped or table dropped during query.
+    * See `Failure Resolution`_ for automatic resolving query failures.
```

Comment:
typo: "for automatic resolving query failures" -> "for auto-resolving of query failures"